### ISSUE TITLE
GraphQL Datastore codegen model field member mismatch

### ISSUE THREAD

**Issue by maceip** [aws-amplify/amplify-swift#2644]
### Describe the bug

xcode 14.0.1
amplify 10.5.2
amplify swift 1.x and 2.x

with this schema (linked from https://github.com/aws-amplify/amplify-ios-samples/blob/179d54c1333ff1a1a3830ab366520a312118b87d/Photo-Sharing-Sample/schema.graphql):

```swift
type Post
    @model
    @auth(rules: [{ allow: owner, operations: [create, delete] }]) {
    id: ID!
    postBody: String!
    pictureKey: String!
    createdAt: AWSDateTime!
    postedBy: User! @belongsTo
}

type User
    @model
    @auth(rules: [{ allow: owner, operations: [create, delete] }]) {
    id: ID!
    username: String!
    profilePic: String!
    posts: [Post] @hasMany(fields: ["id"])
}
```
when doing 

```swift
:Amplify.DataStore.save(User(id: "\(authUser.userId)",
                        username: authUser.username,
                        profilePic: "emptyUserPic"))
```

I get the following error:

2022-12-20 22:42:59.062248-0400 PhotoSharingSample[75767:4147505] [ProcessMutationErrorFromCloudOperation] GraphQLError missing extensions and errorType GraphQLError(message: "**The variables input contains a field name \'posts\' that is not defined for input object type \'CreateUserInput\' ", locations: nil, path: nil, extensions: nil)**
2022-12-20 22:42:59.062672-0400 PhotoSharingSample[75767:4147505] [Amplify] The operation couldn’t be completed. (Amplify.DataStoreError error 0.)

I've tried to edit the schema or the generated swift files directly to prevent including the 'posts' field to the generated 'CreateUserInput' object, but not sure how. 

any ideas?

### Steps To Reproduce

```swift
git clone https://github.com/aws-amplify/amplify-ios-samples.git
cd amplify-ios-samples/Photo-Sharing-Sample
amplify init
amplify add auth (cognito)
amplify add api (with conflict detection-> auto merge)
amplify add storage
amplify codegen models
amplify push
```

### Expected behavior

no errors

### Amplify Framework Version

main

### Amplify Categories

API, DataStore

### Dependency manager

Swift PM

### Swift version

swift-driver version: 1.62.15 Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51) Target: arm64-apple-macosx13.0

### CLI version

10.5.2

### Xcode version

Xcode 14.1 Build version 14B47b

### Relevant log output

```shell
2022-12-20 22:42:59.062248-0400 PhotoSharingSample[75767:4147505] [ProcessMutationErrorFromCloudOperation] GraphQLError missing extensions and errorType GraphQLError(message: "The variables input contains a field name \'posts\' that is not defined for input object type \'CreateUserInput\' ", locations: nil, path: nil, extensions: nil)
2022-12-20 22:42:59.062672-0400 PhotoSharingSample[75767:4147505] [Amplify] The operation couldn’t be completed. (Amplify.DataStoreError error 0.)

2022-12-20 23:22:30.666685-0400 PhotoSharingSample[78129:4181911] [AWSDataStorePlugin] StorageEngine completed with error: DataStoreError: One or more errors occurred syncing models. See below for detailed error description.
Recovery suggestion: DataStoreError: An error occurred syncing Post
Caused by:
DataStoreError: failed to process graphqlResponseData
Caused by:
APIError: failed to process graphqlResponseData
Caused by:
DataStoreError: The key `__typename` was not found
Recovery suggestion: Check if the parsed JSON contains the expected `__typename`
Caused by:
DataStoreError: One or more errors occurred syncing models. See below for detailed error description.
Recovery suggestion: DataStoreError: An error occurred syncing Post
Caused by:
DataStoreError: failed to process graphqlResponseData
Caused by:
APIError: failed to process graphqlResponseData
Caused by:
DataStoreError: The key `__typename` was not found
Recovery suggestion: Check if the parsed JSON contains the expected `__typename`
```

### Is this a regression?

No

### Device

iphone 14 simulator

### iOS Version

16.1

**Comment by maceip** [comment]
these errors go away using amplify version 7.6.2; not sure what all has changed since then...

**Comment by sebsto** [comment]
I have a demo app that used to be working. Now one entity is not syncing, it fails on a @hasMany relationship, just like described above

**Platform** : iOS - Swift
**Amplify Lib Version** 2.2.1
**Amplify CLI version** : 10.6.1
**Schema** : https://github.com/sebsto/amplify-datastore-swift/blob/main/schema.graphql
**Model** : Podcast 1 - n Episodes
`podcast` model has an `episodes` field.  The field is present in the Swift generated model files.  The field is not defined in the generated GraphQL schema (which seems reasonable)

Error when saving entity:
```
2023-01-10 09:49:34.353757+0100 DataStoreDemo[7988:7963329] [SyncMutationToCloudOperation] GraphQL mutation operation received result: success(Swift.Result<AWSPluginsCore.MutationSync<AWSPluginsCore.AnyModel>, Amplify.GraphQLResponseError<AWSPluginsCore.MutationSync<AWSPluginsCore.AnyModel>>>.failure(GraphQLResponseError<MutationSync<AnyModel>>: GraphQL service returned a successful response containing errors: [Amplify.GraphQLError(message: "The variables input contains a field name \'episodes\' that is not defined for input object type \'CreatePodcastDataInput\' ", locations: nil, path: nil, extensions: nil)]
2023-01-10 09:42:43.679449+0100 DataStoreDemo[6565:7951807] [Amplify] The operation couldn't be completed. (Amplify.DataStoreError error 0.)
..
2023-01-10 09:49:34.357333+0100 DataStoreDemo[7988:7963333] [ProcessMutationErrorFromCloudOperation] GraphQLError missing extensions and errorType GraphQLError(message: "The variables input contains a field name \'episodes\' that is not defined for input object type \'CreatePodcastDataInput\' ", locations: nil, path: nil, extensions: nil)
```

**Comment by sebsto** [comment]
@maceip what is amplify 7.6.2 ? Are you refering to an old version of the CLI ?

**Comment by sebsto** [comment]
I just tested a couple of older Amplify CLI versions
10.5.0 : not working
10.3.0 : working

So the regression appears in between 10.3.0 and 10.5.0

**Comment by lawmicha** [comment]
There shouldn't be any changes on the CLI / backend provisioning since the input to the create mutation request has not changed. This is a library issue that either wasn't caught with the CPK testing that existing use cases still work as expected or this was a regression that was introduced in one of the latest commits. There was a change recently close to the changes in the PR that fixes the problem here: https://github.com/aws-amplify/amplify-swift/commit/3a91358b14d2e9f9d51a6c03b89c467a9d8390e3 .

**Comment by sebsto** [comment]
I confirm the fix [on this branch](https://github.com/aws-amplify/amplify-swift/pull/2661/) works for me.
I tested it with Amplify CLI 10.6.1

Thank you @lawmicha !

**Comment by atierian** [comment]
Fix included in releases 2.3.0 and 1.29.0
Please comment here or open a new issue if you still run into this issue after updating.
